maintenance/wikia/cleanEventsLocalUsers.php

```diff
@@ -6,10 +6,62 @@
  * Remove incorrect entries from specials.events_local_users
  *
  * @see PLATFORM-2061
+ * @see PLATFORM-2094
  */
 class CleanEventsLocalUsersMaintenance extends Maintenance {
 
+	const TABLE_NAME = 'events_local_users';
+
 	public function execute() {
+		$this->removeIncorrectUsers();
+		#$this->removeBogusRows();
+	}
+
+	public function removeIncorrectUsers() {
+		global $wgCityId, $wgSpecialsDB;
+		$dbw = $this->getDB( DB_MASTER, [], $wgSpecialsDB );
+
+		// select user_id, user_name from events_local_users where wiki_id = 878889;
+		$res = $dbw->select(
+			self::TABLE_NAME,
+			'user_id, user_name',
+			[
+				'wiki_id' => $wgCityId,
+				'user_id > 0'
+			],
+			__METHOD__
+		);
+
+		while ( $row = $res->fetchRow() ) {
+			$user_name = User::newFromId( $row['user_id'] )->getName();
+
+			// user_id and user_name do not match, remove this entry
+			if ( $user_name !== $row['user_name'] ) {
```

Comment:
The row will be removed, but the script that generates this data will put it back (using data from local `revision` table).